src/WebJobs.Script/Workers/Rpc/Configuration/EnvironmentCondition.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    public class EnvironmentCondition : IWorkerProfileCondition
+    {
+        private readonly ILogger _logger;
+        private readonly ISystemRuntimeInformation _systemRuntimeInformation;
+        private readonly IEnvironment _environment;
+
+        public EnvironmentCondition(ILogger logger, ISystemRuntimeInformation systemRuntimeInformation, IEnvironment environment, string name, string expression)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _systemRuntimeInformation = systemRuntimeInformation ?? throw new ArgumentNullException(nameof(systemRuntimeInformation));
+            Type = ConditionType.Environment;
+            Name = name;
+            Expression = expression;
+            Validate();
```

Comment:
Is it right to call `Validate();` condition here?

Reply:
Calling this here mean that we every time we instantiate a new condition, it would be validated. Ideally this should only happen once so I think it's fine to call here. The only other place I can think of would be at the factory level maybe. But I think this is fine - @fabiocav ?